pkg/scheduler/internal/cache/cache.go

```diff
@@ -63,6 +63,8 @@ type schedulerCache struct {
 	nodeTree  *NodeTree
 	// A map from image name to its imageState.
 	imageStates map[string]*imageState
+	// A map from node label to node names
+	toplogyInfo schedulernodeinfo.TopologyInfo
```

Comment:
Do we really need to put this in the cache? Basically, we are building and maintaining this structure in the scheduler memory even if no pods in the cluster use inter-pod affinity. This increases scheduler's memory usage. Is there any strong reason for having it here?

Reply:
The reason is that if we update the (global) topologyInfo only upon scheduling of an inter-affinity pod, it will cause an instant O(n) time. In current design, it's an average cost on each operation (update on node labels).

Comparing to the average time cost, the consistent space cost (memory footprint you pointed out) more concerns me. Let me see if it can be optimized.